### ISSUE TITLE
Empty Button and Broken ARIA Menu

### ISSUE THREAD

**Issue by emleddin** [rstudio/rmarkdown#1943]
I originally posted a [question](https://community.rstudio.com/t/empty-button-and-broken-aria-menu/86730) in the RStudio Community, but I will include the relevant details here.

I created a website with RMarkdown, and the code is on my [GitHub](https://github.com/emleddin/rmarkdown-thrive-test). When I ran an accessibility check using [WAVE from WebAim](https://wave.webaim.org/), the site failed for two reasons:

1. Empty Button
2. Broken ARIA Menu

I found a way to fix for these issues in the straight HTML, but those changes have to be applied to each page, and anytime the website is rebuilt, those changes will get overwritten. 

**Question**: Is there a way to change the root HTML and fix it for these accessibility errors?

**Empty Button Error**

Original code:
```
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
```

Fix (adding a `sr-only` span):
```
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
```

**Broken ARIA Menu Error**

Original code:
```
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Resources

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="resources.html">Crisis Resources</a>
    </li>
    <li>
      <a href="reading-list.html">Reading List</a>
    </li>
    <li>
      <a href="STEMM-orgs.html">STEMM Organizations</a>
    </li>
  </ul>
</li>
```

Fix (removing `aria-expanded` and the `button` and `menu` roles):
```
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown">
    Resources

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu">
    <li>
      <a href="resources.html">Crisis Resources</a>
    </li>
    <li>
      <a href="reading-list.html">Reading List</a>
    </li>
    <li>
      <a href="STEMM-orgs.html">STEMM Organizations</a>
    </li>
  </ul>
</li>
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.
